modules/ROOT/pages/websocket/websocket-connector-triggers.adoc

```diff
@@ -0,0 +1,164 @@
+
+You can only have one On Inbound Connection trigger for a given path. To use this trigger, you must also configure a matching <<inbound-listener-trigger,Inbound Listener>> for the path.
+
+=== Sending Responses
+
+The app flow associated with the On Inbound Connection trigger is invoked after the WebSocket is established. Therefore the WebSocket might already be broadcasting or receiving messages. The messages are queued and not delivered to the flow until the event triggered by the On Inbound Connection trigger completes. If the event fails, all queued messages are discarded and the connection drops.
+
+=== Example
+
+The following app flow validates attributes from the originating request and closes the connection if the validation doesn't pass:
+
+[source,xml,linenums]
+----
+<flow name="onNewInboundConnection">
+    <websocket:on-inbound-connection doc:name="On Inbound Connection" config-ref="WebSockets_Config" path="/chat"/>
+    <choice doc:name="Choice" >
+        <when expression="#[contains(attributes.headers.alias, 'admin')]">
+            <websocket:close socketId="#[attributes.socketId]" reason="Cannot impersonate the admin" config-ref="WebSockets_Config"/>
+        </when>
+        <otherwise >
+            <logger level="INFO" doc:name="Logger" message="Successfully connected to WebSocket #[attributes.socketId]" />
+        </otherwise>
+    </choice>
+</flow>
+<flow name="chat">
+    <websocket:inbound-listener doc:name="Inbound listener" path="/chat" config-ref="WebSockets_Config"/>
+    <logger level="INFO" doc:name="Logger" message="#['Received message' ++ payload]"/>
+</flow>
+----
+
+=== Output
+
+The On Inbound Connection trigger consists of:
+
+* An empty message payload.
+* An object of type `InboundWebSocketAttributes` that contains message attributes.
+
+== On Socket Close Trigger [[on-websocket-close-trigger]]
+
+Use the `<websocket:on-socket-close` trigger to perform cleanup tasks when a WebSocket closes, such as logging and updating the app state.
+
+The flow associated with the Socket Close trigger is invoked after the WebSocket closes. Therefore, the triggered flow cannot send messages through the WebSocket.
+
+=== Example
+
+The following example closes the WebSocket connection identified by the `ws` socket ID:
+
+*Reviewers: Can we show a bigger example here, such as one that also logs a message when the socket is closed*
+
+----
+<websocket:on-connection-closed path="/ws/*" config-ref="ws" />
+----
+
+=== Output
+
+The On Socket close trigger returns:
+
+* An empty message payload.
+* An object of type `CloseSocketEvent` that contains the message attributes.
+
+== Outbound Listener [[outbound-listener-trigger]]
+
+Use the `<websocket:outbound-listener>` trigger to invoke an app flow when a message is received on a client's outbound endpoint. You can use the associated flow to perform tasks such as sending a return message to the server or processing the information sent by the server.
```

Comment:
`is received on a client's outbound endpoint.` -> `is received through an outbound WebSocket`